lib/pagerduty/http_transport.rb

```diff
@@ -4,25 +4,31 @@
 
 class Pagerduty
   # @api private
-  module HttpTransport
+  class HttpTransport
```

Comment:
Change this transport from a non-instantiated module to a class so we can store the proxy configuration as immutable state on an instance.